`dump(none` triggers compiler bug

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.3.2 6e46933.e5d1881
**OS:** linux, Ubuntu 22.04.1 LTS (VM)

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?** Executed the following code:
```v
fn main() {
	dump(none)
}
```

**What did you expect to see?** Should have printed `<none>`


**What did you see instead?**
```
$ v run bug.v                
==================
/tmp/v_1001/bug.17008390786881131346.tmp.c:12122: error: cannot convert 'extern struct IError' to 'struct none'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.
```

--- a/vlib/v/gen/c/dumpexpr.v
+++ b/vlib/v/gen/c/dumpexpr.v
@@ -45,7 +45,12 @@ fn (mut g Gen) dump_expr_definitions() {
 		deref, _ := deref_kind(str_method_expects_ptr, is_ptr, dump_type)
 		to_string_fn_name := g.get_str_fn(typ.clear_flag(.shared_f))
 		ptr_asterisk := if is_ptr { '*'.repeat(typ.nr_muls()) } else { '' }
-		mut str_dumparg_type := g.cc_type(dump_type, true) + ptr_asterisk
+		mut str_dumparg_type := ''
+		if dump_sym.kind == .none_ {
+			str_dumparg_type = 'IError' + ptr_asterisk
+		} else {
+			str_dumparg_type = g.cc_type(dump_type, true) + ptr_asterisk
+		}
 		if dump_sym.kind == .function {
 			fninfo := dump_sym.info as ast.FnType
 			str_dumparg_type = 'DumpFNType_$name'
@@ -67,6 +72,8 @@ fn (mut g Gen) dump_expr_definitions() {
 		surrounder.add('\tstring sline = int_str(line);', '\tstring_free(&sline);')
 		if dump_sym.kind == .function {
 			surrounder.add('\tstring value = ${to_string_fn_name}();', '\tstring_free(&value);')
+		} else if dump_sym.kind == .none_ {
+			surrounder.add('\tstring value = _SLIT("none");', '\tstring_free(&value);')
 		} else {
 			surrounder.add('\tstring value = ${to_string_fn_name}(${deref}dump_arg);',
 				'\tstring_free(&value);')

--- a/vlib/v/tests/inout/dump_none.out
+++ b/vlib/v/tests/inout/dump_none.out
@@ -1,0 +1,1 @@
+[vlib/v/tests/inout/dump_none.vv:2] none: none

--- a/vlib/v/tests/inout/dump_none.vv
+++ b/vlib/v/tests/inout/dump_none.vv
@@ -1,0 +1,3 @@
+fn main() {
+	dump(none)
+}

